src/D3MHub.sol

```diff
@@ -219,170 +219,123 @@ contract D3MHub {
         emit File(ilk, what, data);
     }
 
-    // --- Deposit controls ---
-    function _wind(bytes32 ilk, ID3MPool _pool, uint256 amount) internal {
-        // IMPORTANT: this function assumes Vat rate of D3M ilks will always be == 1 * RAY (no fees).
-        // That's why this module converts normalized debt (art) to Vat DAI generated with a simple RAY multiplication or division
-        // This module will have an unintended behaviour if rate is changed to some other value.
-        if (amount == 0) {
-            emit Wind(ilk, 0);
-            return;
-        }
+    // --- Internal functions that are called from exec(bytes32 ilk) ---
 
-        vat.slip(ilk, address(_pool), int256(amount));
-        vat.frob(ilk, address(_pool), address(_pool), address(this), int256(amount), int256(amount));
-        // normalized debt == erc20 DAI (Vat rate for D3M ilks fixed to 1 RAY)
-        daiJoin.exit(address(_pool), amount);
-        _pool.deposit(amount);
+    function _wipe(bytes32 ilk, ID3MPool _pool, address urn) internal {
+        uint256 amount = _pool.maxWithdraw();
+        if (amount > 0) {
+            _pool.withdraw(amount);
+            daiJoin.join(address(this), amount);
+            vat.move(address(this), vow, amount * RAY);
 
-        emit Wind(ilk, amount);
+            uint256 toSlip = _min(vat.gem(ilk, urn), amount);
+            // amount bounds toSlip and amount * RAY bounds amount to be much less than MAXINT256
+            vat.slip(ilk, urn, -int256(toSlip));
+            emit Unwind(ilk, amount);
```

Comment:
Do we want a `Wipe` event here instead of `Unwind`? Might be good to avoid ambiguity.

Reply:
Hmmm my personal opinion is `Unwind` is just fine, as they can only happen in different stages of the contract (meaning that there is certainty on which one was used), and also as in fact they are doing the exact same thing.
It is not like within `wipe` we will clean the whole CDP for sure, we are just unwinding with a different way to calculate the amount.
But will leave this open to the rest to give their opinion.

Reply:
I like `Unwind` as (like gonza pointed out) we might not be able to wipe completely

Reply:
I was just thinking it would be good to clear which code path had been triggered...but don't feel too strongly.